### ISSUE TITLE
Missing signature argument to addFunction ‘__emval_new’ when compiled as MAIN_MODULE

### ISSUE THREAD

**Issue by DoneListen** [emscripten-core/emscripten#19012]
emcc -v:
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.0 (8e1e305519e1027726a48861a1fec5662f7e18a2)
clang version 14.0.0 (https://github.com/llvm/llvm-project 1a929525e86a20d0b3455a400d0dbed40b325a13)
Target: wasm32-unknown-emscripten
Thread model: posix
InstalledDir: D:\SvnCode\emsdk-main\upstream\bin

I used -s MAIN MODULE=2 to compile the argument，but Console error：
```
Aborted(Assertion failed: Missing signature argument to addFunction: function __emval_new(handle, argCount, argTypes, args) {
 handle = Emval.toValue(handle);
 var newer = emval_newers[argCount];
 if (!newer) {
  newer = craftEmvalAllocator(argCount);
  emval_newers[argCount] = newer;
 }
 return newer(handle, argTypes, args);
})
```

**Comment by DoneListen** [comment]
I compiled the debug version， has this problem.
I found the reason because debug exports this method ”_emval_new“, Why is it exported.
![a16eac7ccb505144a47df49f13544280](https://user-images.githubusercontent.com/91657327/230534699-53541e3e-4e70-4bee-ad6f-e1c411b5c24c.png)
![xio](https://user-images.githubusercontent.com/91657327/230534707-661899fc-9e70-48ef-a3ce-69504285765f.png)

**Comment by DoneListen** [comment]
@sbc100 @kripken

**Comment by DoneListen** [comment]
I have found the cause of the problem，I used ”val::global("XMLHttpRequest").new_()“ in my code. When I compile the debug version the browser then loads with an error like this. How should this be resolved. 
The release version does not have this problem.
@sbc100 @kripken

**Comment by sbc100** [comment]
Yes, I believe this was fixed in #17075, which was released in v3.1.12.
